test/IntegrationTests/SmokeTests.cs

```diff
@@ -89,6 +89,28 @@ public void ApplicationIsExcluded()
         VerifyTestApplicationNotInstrumented();
     }
 
+#if NET6_0_OR_GREATER
+    [Fact]
+    [Trait("Category", "EndToEnd")]
+    public void ApplicationFailFast()
```

Comment:
Should we also add a test that confirms that we do not crash the application?

Reply:
Few lines above there is a test ApplicationIsExcluded which is doing exactly what you are asking for.

Should I add here some comment refering this test?

Reply:
I think that it is not intuitive that the `ApplicationIsExluded` test covers that scenario. I think that a comment is necessary (at a minimum). Ideally we would have a separate test so that the 2 tests can evolve independently.

Reply:
New test added. For now, the content is exactly the same as for old test, but I like the idea to be open for evolving.